bindings/rust/wasmedge-sys/src/config.rs

```diff
@@ -95,99 +145,273 @@ impl Config {
     }
 
-    /// Set the generic binary option of AOT compiler.
-    pub fn generic_binary(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureCompilerSetGenericBinary(self.ctx, enable) };
+    /// Sets the generic binary option of AOT compiler.
+    pub fn enable_generic_binary(self, flag: bool) -> Self {
+        unsafe { wasmedge::WasmEdge_ConfigureCompilerSetGenericBinary(self.ctx, flag) };
         self
     }
 
-    /// Get the generic binary option of AOT compiler.
+    /// Checks if the generic binary option of AOT compiler turns on or not.
     pub fn is_generic_binary(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureCompilerIsGenericBinary(self.ctx) }
     }
 
-    /// Enable or disable instruction counting.
-    pub fn count_instructions(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetInstructionCounting(self.ctx, enable) };
+    /// Sets the instruction counting option.
+    pub fn enable_instruction_counting(self, flag: bool) -> Self {
+        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetInstructionCounting(self.ctx, flag) };
         self
     }
 
-    /// Get the instruction counting option.
+    /// Checks if the instruction counting option turns on or not.
     pub fn is_instruction_counting(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureStatisticsIsInstructionCounting(self.ctx) }
     }
 
-    /// Enable or disable cost cost measuring.
-    pub fn measure_costs(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetCostMeasuring(self.ctx, enable) };
+    /// Sets the cost measuring option.
+    pub fn enable_cost_measuring(self, flag: bool) -> Self {
+        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetCostMeasuring(self.ctx, flag) };
         self
     }
 
-    /// Get the cost measuring option.
+    /// Checks if the cost measuring option turns on or not.
     pub fn is_cost_measuring(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureStatisticsIsCostMeasuring(self.ctx) }
     }
 
-    /// Set the time measuring option.
-    pub fn measure_time(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetTimeMeasuring(self.ctx, enable) };
+    /// Sets the time measuring option.
+    pub fn enable_time_measuring(self, flag: bool) -> Self {
+        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetTimeMeasuring(self.ctx, flag) };
         self
     }
 
-    /// Get the cost measuring option.
+    /// Checks if the cost measuring option turns on or not.
     pub fn is_time_measuring(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureStatisticsIsTimeMeasuring(self.ctx) }
     }
 }
 
-// # TODO: WasmEdge_HostRegistration
+/// Code Generate for `WasmEdge_Proposal` :
+///
+/// ```rust
+/// impl Config {
+///     pub fn enable_bulkmemoryoperations(self, enable: bool) -> Self {
```

Comment:
I am not sure if we should use `enable_` and `disable_` or `set_`. This is very confusing.

Reply:
This comment and the subsequent macro were designed by other contributors. My suggestion is to keep them as they are now. After this round of the doc comment task is done, we plan to initiate a new task to review all APIs of Rust bindings, that could refine/remove some previously introduced design. Thanks!

Reply:
Okay.